### PR TITLE
Better target filtering and command handling

### PR DESCRIPTION
**Pull request by billti**
Show all targets and providers (if not explicitly excluded), and only allow job submission on QIR-supporting targets, and downloads for jobs with results.

### CODE REVIEW

File: vscode/src/azure/commands.ts [hunk 1]
```diff
@@ -19,19 +19,62 @@ import {
...
+
 export function initAzureWorkspaces() {
   const workspaceTreeProvider = new WorkspaceTreeProvider();
-  vscode.window.createTreeView("quantum-workspaces", {
+  const treeView = vscode.window.createTreeView("quantum-workspaces", {
     treeDataProvider: workspaceTreeProvider,
   });
 
+  treeView.onDidChangeSelection((e) => {
+    // Capture the selected item, and set context if if supports job submission or results download.
```
Comment (idavis) [inline vscode/src/azure/commands.ts:35]:
```suggestion
    // Capture the selected item and set context if the supports job submission or results download.
```

File: vscode/src/azure/commands.ts [hunk 2]
```diff
@@ -19,19 +19,62 @@ import {
...
+import { targetSupportQir } from "./providerProperties";
 
 const corsDocsUri = "https://github.com/microsoft/qsharp/wiki/Enabling-CORS";
 
+let currentTreeItem: WorkspaceTreeItem | undefined = undefined;
+
 export function initAzureWorkspaces() {
   const workspaceTreeProvider = new WorkspaceTreeProvider();
-  vscode.window.createTreeView("quantum-workspaces", {
+  const treeView = vscode.window.createTreeView("quantum-workspaces", {
```
Comment (minestarks) [inline vscode/src/azure/commands.ts:30]: Didn't catch this in the first go around, but this (and other resources) should probably be added to `context.subscriptions` to be disposed of when the extension unloads.

Same with the return value of `onDidChangeSelection`.

File: vscode/src/azure/commands.ts [hunk 3]
```diff
@@ -19,19 +19,62 @@ import {
...
+      supportsQir
+    );
+    vscode.commands.executeCommand(
+      "setContext",
+      "qsharp-vscode.treeItemSupportsDownload",
+      supportsDownload
+    );
+  });
+
   vscode.commands.registerCommand(
```
Comment (minestarks) [inline vscode/src/azure/commands.ts:69]: `registerCommand` also returns `Disposable`.

File: vscode/package.json [hunk 4]
```diff
@@ -142,14 +158,13 @@
...
-        "title": "Download full Azure Quantum job results",
+        "title": "Download Azure Quantum job results",
         "icon": "$(cloud-download)"
       },
       {
         "command": "qsharp-vscode.setTargetProfile",
         "category": "Q#",
-        "title": "Set the Azure Quantum QIR target profile",
-        "enablement": "resourceLangId == qsharp"
+        "title": "Set the Azure Quantum QIR target profile"
```
Comment (minestarks) [inline vscode/package.json:167]: Why did we remove the enablement filter on this?

**Review by minestarks — APPROVED** [review]

**Review by idavis — APPROVED** [review]

File: vscode/package.json (hunk 4 above)
Comment (billti) [reply vscode/package.json:167]: It's a global setting (saved in your user settings), not document specific. I'm not sure you should have to have a Q# file open to be able to change it from the menu. I could be persuaded either way though.

**Pull request merged**